### ISSUE TITLE
Lazy plotting

### ISSUE THREAD

**Issue by rafaqz** [rafaqz/GeoData.jl#172]
We should be able to plot larger than memory files and zoom in to get detail.

See: https://github.com/yeesian/ArchGDAL.jl/issues/195

**Comment by rafaqz** [comment]
This works now, but not with zooming: for that we need Makie.jl recipes.

**Comment by felixcremer** [comment]
With PyramidScheme we have the interactive plotting after a short preparation time for computing the pyramid. Would you expect this to also be able on the fly?

**Comment by rafaqz** [comment]
How easy would it be to hook PyramidScheme into Rasters?

**Comment by felixcremer** [comment]
I am actually already using Rasters for the Makie recipe. And the Readme shows an example of computing the pyramids of a raster from RasterDataSources. This works because DiskArrayEngine uses views an doesn't need a special type anymore.

The next step is to register diskarrayengine and pyramidscheme and to streamline the API.

**Comment by felixcremer** [comment]
Once the registration is done, we should add a small example of using PyramidScheme.jl to the Rasters docs.

**Comment by felixcremer** [comment]
PyramidScheme is registered now and this could get a mention in the Rasters docs. 
There is an example with a RasterDataSource in the PyramidScheme README which could be used.

**Comment by rafaqz** [comment]
Should show that at JuliaCon too

**Comment by felixcremer** [comment]
I am going to show PyramidScheme and building a Pyramid from a Raster at my PyramidScheme.jl talk tomorrow.

**Comment by rafaqz** [comment]
Ok I won't do that in the rasters talk then!
